lib/identifierMatcher.js

```diff
@@ -0,0 +1,88 @@
+//lhs => specific
+//rhs => general
+const matchScore = (lhs, rhs) => {
+    const [lhsPath, lhsToken] = getPathAndToken(lhs)
+    const [rhsPath, rhsToken] = getPathAndToken(rhs)
+
+    if (lhsToken !== rhsToken) {
+        return 0;
+    }
+
+    if (rhsPath.length === 0) {
+        return 1;
+    }
+
+    if (lhsPath.length === 0) {
+        return 0;
+    }
+
+    if (lhs === rhs) {
```

Comment:
<possible misunderstanding of Javascript>

Can this be `==` instead of `===`? If they are different instances but still the same content then they should be considered the same. Or is Javascript being "special" with its meaning of `===`?

Reply:
It's "special"; `===` in js is for strict equality (not pointer comparison); `==` would match silly things like `"" == 0` or `false == 0`. Im not sure if it's a problem here but I went with "strict"

Reply:
Fair enough - worst case we compare two identical values the long way instead of short-cutting - but I think you're right and this does what you (but apparently not I!) expect.